src/main/java/com/comodide/views/GraphExportView.java

```diff
@@ -0,0 +1,124 @@
+package com.comodide.views;
+import com.mxgraph.util.mxCellRenderer;
+import com.mxgraph.swing.mxGraphComponent;
+import com.comodide.helper.GraphComponentHolder;
+
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.event.ActionEvent;
+import java.io.File;
+import java.io.IOException;
+import java.awt.image.BufferedImage;
+import java.awt.event.ActionListener;
+
+import javax.imageio.ImageIO;
+import javax.swing.*;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JFileChooser;
+
+import org.protege.editor.owl.ui.view.AbstractOWLViewComponent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GraphExportView extends AbstractOWLViewComponent
+{
+    private static final Logger log              = LoggerFactory.getLogger(EdgeInspectorView.class);
+    //using singleton class instance to get graph component of BasicGraphEditor class
+    GraphComponentHolder obj=GraphComponentHolder.getInstance();
```

Comment:
There's no reason for this to have default visibility rather than private, and this data should certainly be a static reference rather than giving each object a separate reference to the same singleton that they must each store separately.
```
private static final GraphComponentHolder graphComponentHolder = GraphComponentHolder.getInstance();
```
Also, there are two variables, `obj` and `graphComponent`, but `obj` is never used except to initialize `graphComponent`, so why not just have one of them?

Actually, though, the biggest issue here is the risk of falling out of sync with the singleton, because you extract the `graphComponent` and store it in the View, but this doesn't get updated when the singleton has its component changed, meaning that the `GraphExportView` can be holding a reference to the wrong component.

To fix this, just do not store a reference to `graphComponent`, but rather store only the reference to the holder (your `obj`), and just access the component when you need it. Better yet, you can get rid of the singleton class entirely and store the graphComponent directly here in this class (or the ComodideEditor class) as a private static member, and then you won't have any such issue. If you must use a singleton for some reason, you could also avoid these issues by just calling `getInstance` any time you need to access the instance.

Reply:
Hi @JonIsPatented, have done the requested changes and removed singleton class. Defined private static member in GraphExportView and a public method to set its value in ComodideEditor class